web/src/presence.ts

```diff
@@ -304,6 +322,48 @@ export function last_active_date(user_id: number): Date | undefined {
     return new Date(info.last_active * 1000);
 }
 
+export function map_status(status: string): "active" | "idle" | "offline" {
+    switch (status) {
+        case "active":
+        case "idle":
+        case "offline":
+            return status;
+        default:
+            return "offline";
+    }
+}
+
+export const fetch_user_last_seen_status = (user_id: number, callback: () => void): void => {
+    const url = `json/users/${user_id}/presence`;
+
+    channel.get({
+        url,
+        success(data: unknown) {
+            const parsed_data = user_last_seen_response_schema.safeParse(data);
+
+            if (parsed_data.success) {
+                const response = parsed_data.data;
+
+                if (response.result === "success" && response.presence) {
+                    const {aggregated} = response.presence;
+                    presence_info.set(user_id, {
+                        status: map_status(aggregated.status),
```

Comment:
Why is the `map_status` function needed? It just returns the passed string.

Reply:
The type of `aggregated.status` is string and the status of `presence_info.set` expected some value which must be `"Active" | "Idle" | "Offline"`